local-api.js

```diff
@@ -1,40 +1,43 @@
+'use strict'
+const Permissions = require('./permissions')
+const u = require('./util')
 
-var Permissions  = require('./permissions')
-var u            = require('./util')
-
-module.exports = 
-
-function createLocalCall(api, manifest, perms) {
+module.exports = function createLocalCall (api, manifest, perms) {
   perms = Permissions(perms)
 
-  function has(type, name) {
+  function has (type, name) {
     return type === u.get(manifest, name)
   }
 
-  function localCall(type, name, args) {
-
-    if(name === 'emit')
+  function localCall (type, name, args) {
+    if (name === 'emit') {
       throw new Error('emit has been removed')
-
-    //is there a way to know whether it's sync or async?
-    if(type === 'async')
-      if(has('sync', name)) {
-        var cb = args.pop(), value
-        try { value = u.get(api, name).apply(this, args) }
-        catch (err) { return cb(err) }
+    }
+
+    // is there a way to know whether it's sync or async?
+    if (type === 'async') {
+      if (has('sync', name)) {
+        const cb = args.pop()
+        let value
+        try {
+          value = u.get(api, name).apply(this, args)
```

Comment:
the return a little further down can be move in here because the catch returns. This also allows you to move value in this branch. As I write this I wonder if that is those other improvements you had in mind?

Reply:
I thought about this, but I was worried that the `try` would cover errors that occur **inside** `cb`, and that would change the behavior of muxrpc. I think the author's intent here was correct: to catch errors that happen exclusively in the `u.get()` and the "muxrpc API" function call, but not catch errors elsewhere.

Reply:
right